src/realm/row.hpp

```diff
@@ -186,14 +193,17 @@ class RowFuncs {
 template <class T>
 class BasicRowExpr : public RowFuncs<T, BasicRowExpr<T>> {
 public:
-    BasicRowExpr() noexcept;
+    BasicRowExpr() noexcept = default;
 
     template <class U>
     BasicRowExpr(const BasicRowExpr<U>&) noexcept;
 
+    template <class U>
+    BasicRowExpr(const BasicRow<U>&) noexcept;
```

Comment:
BasicRow has a move constructor, but it still has to acquire a lock so it isn't actually meaningfully cheaper than copying a Row. The object store uses RowExpr for the same thing as core (returning temporaries).